vpn/dylib/lib.go

```diff
@@ -0,0 +1,93 @@
+//go:build darwin
+// +build darwin
+
+package main
+
+import "C"
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"golang.org/x/sys/unix"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"cdr.dev/slog/sloggers/sloghuman"
+
+	"github.com/coder/coder/v2/vpn"
+)
+
+// OpenTunnel creates a new VPN tunnel by `dup`ing the provided 'PIPE'
+// file descriptors for reading, writing, and logging.
+//
+//export OpenTunnel
+func OpenTunnel(cReadFD, cWriteFD int32) int32 {
+	ctx := context.Background()
+
+	conn, err := newFdConn(cReadFD, cWriteFD)
+	if err != nil {
+		return -1
+	}
+
+	// Logs will be sent over the protocol
+	_, err = vpn.NewTunnel(ctx, slog.Make(sloghuman.Sink(io.Discard)), conn)
+	if err != nil {
+		return -1
+	}
+
+	return 0
+}
+
+type pipeConn struct {
```

Comment:
We're going to end up with two implementations of this, so one of us should move their impl to `vpn` or something and export it

Reply:
If you want I can incorporate some of the things in yours like the improved close method into mine and move it (although to keep it compatible with windows you will need to call Dup yourself in this package)